OpenSim/Simulation/Model/ContactGeometry.h

```diff
@@ -33,120 +33,84 @@ namespace OpenSim {
-     */
-    OpenSim::PhysicalFrame& updBody();
-    /**
-     * %Set the Body this geometry is attached to.
-     */
-    void setBody(PhysicalFrame& body);
-    /**
-     * Get the name of the Body this geometry is attached to.
-     */
-    const std::string& getBodyName();
-    /**
-     * %Set the name of the Body this geometry is attached to.  This will cause the
-     * Body to be set to NULL, then resolved when extendConnectToModel() is called.
+    /** This constructor connects this ContactGeometry to the provided `frame`,
+     * and uses the default location and orientation (both `Vec3(0)`).
+     *
+     * @param frame        the PhysicalFrame this geometry is attached to;
      */
-    void setBodyName(const std::string& name);
+    explicit ContactGeometry(const PhysicalFrame& frame);
+
     /**
-     * Get the display_preference of this geometry.
+     * @param location     the location of the geometry expressed in `frame`
+     * @param orientation  the orientation of the geometry expressed in `frame`
+     *                     as XYZ body-fixed Euler angles.
+     * @param frame        the PhysicalFrame this geometry is attached to;
+     *                     this constructor connects this ContactGeometry to
+     *                     the provided `frame`
      */
+    ContactGeometry(const SimTK::Vec3& location,
+                    const SimTK::Vec3& orientation,
+                    const PhysicalFrame& frame);
+
+
+    // ACCESSORS
+    /** Get the PhysicalFrame this geometry is attached to. */
+    const PhysicalFrame& getFrame() const;
+    /** %Set the PhysicalFrame this geometry is attached to. */
+    void setFrame(const PhysicalFrame& body);
+
+    /** Get the display_preference of this geometry. */
     const int getDisplayPreference();
-    /**
-     * %Set the display_preference of this geometry.
-     */
+    /** %Set the display_preference of this geometry. */
     void setDisplayPreference(const int dispPref);
-    /**
-     * Create a new SimTK::ContactGeometry based on this object.
-     */
+
+    /** Create a new SimTK::ContactGeometry based on this object. */
     virtual SimTK::ContactGeometry createSimTKContactGeometry() = 0;
-    /**
-     * Get a Transform representing the position and orientation of the geometry
-     * within the Body it is attached to.
-     */
-    SimTK::Transform getTransform();
+
+    /** Get a Transform representing the position and orientation of the
+     * geometry within the base Frame (e.g., a Body or Ground) it is attached
+     * to (*not* the "frame" that the geometry is connected to). If `B` is the
+     * base or Body Frame, `F` is the PhysicalFrame that this geometry is
+     * connected to, and `P` is the frame that is defined (relative to `F`) by
+     * the location and orientation properties of this class, then this returns
+     * `X_BP = X_BF * X_FP`. */
+    SimTK::Transform findTransformInBaseFrame() const;
```

Comment:
I think you are right, that the geometry could be drawn with a transform to ground instead. But this method is still nice to have when building the [`SimTK::HuntCrossleyForce`](https://github.com/opensim-org/opensim-core/pull/963/files/13bdeebd722bcff3278763b204762c57d10bed09#diff-82843d67d539b05cee170376c728fe38R79) or [`SimTK::ElasticFoundationForce`](https://github.com/opensim-org/opensim-core/pull/963/files/13bdeebd722bcff3278763b204762c57d10bed09#diff-df1955ed2d738ca938e55a7d6d8c7863R74), which I don't think we could specify relative to ground:

```c++
contacts.addBody(set, geom.getFrame().getMobilizedBody(),
                     geom.createSimTKContactGeometry(),
                     geom.findTransformInBaseFrame());
```

Another reason for having this method is that it does nearly the same thing as the previous `ContactGeometry::getTransform()`, reducing the chance that users will introduce a bug into their code when upgrading to 4.0. 

I agree that the name is unfortunate. I would have preferred `findTransformInBody()`. The reason I chose to go with `findTransformInBaseFrame()` was because the base frame may not be an `OpenSim::Body`; it may be `OpenSim::Ground`. I attempted to document this in this method's doxygen.

Considering all this, let me know what you'd like me to do.